nemo/collections/common/tokenizers/gpt2_tokenizer.py

```diff
@@ -0,0 +1,88 @@
+# Copyright 2020 NVIDIA. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# =============================================================================
+
+from typing import Optional
+
+from transformers import GPT2Tokenizer
+
+from nemo.collections.common.tokenizers.tokenizer_spec import TokenizerSpec
+
+__all__ = ['NemoGPT2Tokenizer']
+
+
+class NemoGPT2Tokenizer(TokenizerSpec):
+    def __init__(
+        self,
+        pretrained_model: str,
+        bos_token: Optional[str] = "<|endoftext|>",
+        eos_token: Optional[str] = "<|endoftext|>",
+    ):
+        """
+        Args:
+            pretrained_model: GPT2 pretrained model name from Hugging Face
+            bos_token: the beginning of sequence token
+            eos_token: the end of sequence token.
+        """
+        self.tokenizer = GPT2Tokenizer.from_pretrained(pretrained_model)
+        self.vocab_size = self.tokenizer.vocab_size
+        special_tokens_dict = {}
+        if self.tokenizer.unk_token is None:
+            self.tokenizer.unk_token = "<|unk|>"
+            special_tokens_dict["unk_token"] = "<|unk|>"
+        if self.tokenizer.bos_token is None:
+            special_tokens_dict["bos_token"] = bos_token
+        if self.tokenizer.eos_token is None:
+            special_tokens_dict["eos_token"] = eos_token
+        if self.tokenizer.pad_token is None:
+            special_tokens_dict["pad_token"] = "<|pad|>"
+        self.tokenizer.add_special_tokens(special_tokens_dict)
```

Comment:
could you add a `add_special_tokens` function and all it in the constructor to be consistent with other tokenizers?

Reply:
done